src/ui/components/AbstractToolbar.js

```diff
@@ -1,31 +1,390 @@
+                    const osvcLink = this.props.applicationData.EL_OSVCU
+                                    .replace("{{workOrderId}}", entity.number);
+                    return () => window.open(osvcLink, "_blank");
+                },
+                isDisabled: () => this.props.newEntity,
+                values: {
+                    icon: <Domain/>,
+                    text: "OSVC"
+                }
+            },
+            {
+                key: BUTTON_KEYS.DISMAC,
+                isVisible: () =>  this.props.applicationData.EL_DMUSG &&
+                                  this.props.applicationData.EL_DMUSG.includes(this.props.userGroup),
+                getOnClick: (entityType, entity) => {
+                    const dismacLink = this.props.applicationData.EL_DMURL
+                                                        .replace("{{workOrderId}}", entity.number);
+                    return () => window.open(dismacLink, "_blank");
+                },
+                isDisabled: () => this.props.newEntity,
+                values: {
+                    icon: <Camera/>,
+                    text: "DISMAC"
+                } 
+            },
+            { 
+                key: BUTTON_KEYS.TREC,
+                isVisible: () => {
+                    const entity = this.props.entity;
+                    const { EL_TRWOC } = this.props.applicationData;
+                    return (
+                        EL_TRWOC &&
+                        EL_TRWOC.split(",")
+                            .filter(Boolean)
+                            .includes(entity.classCode)
+                    );
+                },
+                getOnClick: (entityType, entity) => {
+                    const { applicationData } = this.props;
+                    const { EL_TRWRU } = applicationData;
+                    const trecLink = EL_TRWRU.replace(
+                        "{{workOrderId}}",
+                        entity.number
+                    );
+                    return () => window.open(trecLink, "_blank");
+                },
+                isDisabled: () => this.props.newEntity,
+                values: {
+                    icon: <RadiationIcon/>,
+                    text: "TREC"
+                }
+            },
+            {
+                key: BUTTON_KEYS.CREATE_WORKORDER,
+                isVisible: () => true,
+                getOnClick: () => {
+
+                },
+                isDisabled: () => this.props.newEntity,
+                values: {
+                    icon: <WorkorderIcon/>,
+                    text: "Create New Work Order"
+                },
+                getLinkTo: (entity, entityCode) => {
+                    return `/workorder?equipmentCode=${entity.code}`;
+                }
+            }
+        ]
+    }
+
+    generateContent = (viewMode, {key, isVisible, onClick, isDisabled, values, getOnClick, getLinkTo}, entityType, entity) => {
```

Comment:
Could we make this method receive an object, might make things easier so that we do not have to worry about the order:
```javascript
generateContent = ({ viewMode, buttonDefinition, entityType, entity }) => {
...
}

// when calling it
generateContent({
    viewMode: "example",
    buttonDefinition: myButtonDefinitionObjectThatICreatedSomeWhere,
    entityType: myEntityType,
    entity: myEntity,
})
```